### ISSUE TITLE
Add extension point for overriding element Rendering in Node Trees

### ISSUE THREAD

**Issue by skurfuerst** [neos/neos-ui#867]
**[<- must-have for 1.0.0](https://github.com/neos/neos-ui/issues/687)**

One of our core feature sponsors needs a way to manipulate the rendering of the TreeNodes; i.e. he wants to display additional icons based on custom property values.

In order to support that, we at least need to put the tree node React elements in the registry, so they can be overridden from the outside.

**Comment by dimaip** [comment]
Hmm I like the idea, but at the same time we should be careful about creating such ad-hoc APIs as we'd need to maintain them.
Can we maybe declare it as an unplanned extensibility?

**Comment by skurfuerst** [comment]
yes, for sure, I'd definitely declare it as unplanned extensibility!

**Comment by dimaip** [comment]
@skurfuerst can we close it now since you have this fancy extensibility thing in place?

**Comment by mstruebing** [comment]
Not exactly, you'd have to override the whole tree, at the current state it's not possible to only override the tree elements.
I've experienced this while developing this: https://github.com/sitegeist/Sitegeist.TrafficLights

UI-specific code: https://github.com/sitegeist/Sitegeist.TrafficLights/tree/master/Resources/Private/TrafficLights/src

**Comment by dimaip** [comment]
Ok, let's reopen then
